### PR TITLE
feat: lint for iceworks-server

### PR DESCRIPTION
**Pull request by alvinhui**
## Background

- Open lint for iceworks-server;
- Fixed some irregular code with auto-fix function;
- Manually fix some irregular code.

## Notes

- Add `dist/` to .eslintignore, this is the directory of the build results of typescript;
- set `class-methods-use-this` to warning, I think this is a very controversial rule, `this` is not used in many controllers;
- I used `/* eslint */` to ignore some rules for file and `// eslint-disable-line` for line, please check the specific scenario.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const { eslintTS, deepmerge } = require('@ice/spec');
 
 module.exports = deepmerge(eslintTS, {
   rules: {
-    // 禁止用var引入模块：取消
-    "@typescript-eslint/no-var-requires": 0
+    "@typescript-eslint/no-var-requires": 0,
```
Comment (Orange-C) [inline .eslintrc.js]: 规则改动都加上注释说明吧

File: packages/iceworks-server/src/lib/adapter/modules/page/index.ts [hunk 2]
```diff
@@ -137,14 +141,14 @@ export default class Page implements IPageModule {
...
   }
 
-  async getAll(): Promise<IPage[]> {
+  public async getAll(): Promise<IPage[]> {
     const pages = await this.scanPages(this.path);
     return _.orderBy(pages, 'name', 'asc');
   }
 
-  async getOne(): Promise<any> { }
+  public async getOne(): Promise<any> { }
```
Comment (Orange-C) [inline packages/iceworks-server/src/lib/adapter/modules/page/index.ts]: 这些空函数的意义是什么？如果业务没有需要可以删除吧，没必要为了加这个还改写规则

File: packages/iceworks-server/src/lib/adapter/modules/menu/index.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint no-eval: 0 */
```
Comment (Orange-C) [inline packages/iceworks-server/src/lib/adapter/modules/menu/index.ts]: 改成单行取消

File: packages/iceworks-server/src/lib/adapter/modules/page/index.ts (hunk 2 above)
Comment (alvinhui) [reply packages/iceworks-server/src/lib/adapter/modules/page/index.ts]: fixed

File: packages/iceworks-server/src/lib/adapter/modules/menu/index.ts (hunk 3 above)
Comment (alvinhui) [reply packages/iceworks-server/src/lib/adapter/modules/menu/index.ts]: fixed

File: .eslintrc.js (hunk 1 above)
Comment (alvinhui) [reply .eslintrc.js]: 啥意思？我觉得这些规则都比较明确，不需要再注释说明了。
你的意思是加上注释为什么 取消？

**Pull request merged**